### ISSUE TITLE
(x) We should differentiate `isRequired` and `isNullable`

### ISSUE THREAD

**Issue by Fen747** [bluelibs/bluelibs#95]
A bit like with GraphQL, we should, in BluePrint fields, make a difference between `isRequired` and `isNullable`.

`isNullable` => the can possibly be null or simply not exists
`isRequired` => any input MUST have either the value or a default fallback.

There are many cases where a value is required in DB, but not in the form (server-side inference).

We might need to differentiate the different layers `forms` | `graphql` | `database`

**Comment by theodorDiaconu** [comment]
Yes. I fully agree, for the forms let's say you could configure this in the forms, but having required for graphql but not for server not. Do you have an API in mind?

**Comment by Fen747** [comment]
Simple use case : stick to `nexus`-like API : `field.list.string.required.nullable({ ... })`

If we want to differentiate the `required` of the form from the one of the API or DB, then we should probably have it as a field in the option object argument.

**Comment by theodorDiaconu** [comment]
> There are many cases where a value is required in DB, but not in the form (server-side inference).

@Fen747 if you have a field which is required: true, and has a default Value, that default value will be in the form, you can opt out of the form inside the `ui`, on the server side when the model is constructed it will inherit that default value. 

I'm still unsure wether this distinction is necessary, we treat all non required fields as `.nullable()`
